File: backend/app/schemas/critique.py
# backend/app/schemas/critique.py
from pydantic import BaseModel, Field
from typing import List, Optional

class Flag(BaseModel):
    icon: str
    text: str
    severity: str

class Suggestion(BaseModel):
    text: str
    rationale: str

class Citation(BaseModel):
    rule_id: str
    rule_name: str
    arxiv_id: str
    paper_title: str
    authors: str
    year: int | None = None
    url: str
    evidence_from_paper: str | None = None

class Contradiction(BaseModel):
    rule_a: str
    rule_b: str
    type: str
    resolution: str
    llm_reasoning: str

class CritiqueTrace(BaseModel):
    rules_matched: List[str] = Field(default_factory=list)
    contradictions: List[Contradiction] = Field(default_factory=list)
    citations: List[Citation] = Field(default_factory=list)

class CritiqueCard(BaseModel):
    id: str
    material_formula: str
    verdict: str
    score: float
    domain_rules_passed: int
    domain_rules_total: int
    flags: List[Flag] = Field(default_factory=list)
    suggestions: List[Suggestion] = Field(default_factory=list)
    explanation: str
    trace: CritiqueTrace

File: frontend/src/types/critique.ts
export type Flag = {
  icon: string;
  text: string;
  severity: string;
};

export type Suggestion = {
  text: string;
  rationale: string;
};

export type Citation = {
  rule_id: string;
  rule_name: string;
  arxiv_id: string;
  paper_title: string;
  authors: string;
  year: number | null;
  url: string;
  evidence_from_paper?: string | null;
};

export type Contradiction = {
  rule_a: string;
  rule_b: string;
  type: string;
  resolution: string;
  llm_reasoning: string;
};

export type CritiqueTrace = {
  rules_matched: string[];
  contradictions: Contradiction[];
  citations: Citation[];
};

export type CritiqueCard = {
  id: string;
  material_formula: string;
  verdict: string;
  score: number;
  domain_rules_passed: number;
  domain_rules_total: number;
  flags: Flag[];
  suggestions: Suggestion[];
  explanation: string;
  trace: CritiqueTrace;
};
